fastly/platform_tls.go

```diff
@@ -0,0 +1,390 @@
+	Meta    map[string]string   `mapstructure:"meta"`
+	Message string              `mapstructure:"msg"`
+	Data    *PlatformPrivateKey `mapstructure:"data"`
+}
+
+// PlatformPrivateKey .
+type PlatformPrivateKey struct {
+	ID            string                        `mapstructure:"id"`
+	Type          string                        `mapstructure:"type"`
+	Attributes    PlatformPrivateKeysAttributes `mapstructure:"attributes"`
+	Comment       string                        `mapstructure:"comment"`
+	CustomerID    string                        `mapstructure:"customer_id"`
+	CreatedAt     *time.Time                    `mapstructure:"created_at"`
+	UpdatedAt     *time.Time                    `mapstructure:"updated_at"`
+	DeletedAt     *time.Time                    `mapstructure:"deleted_at"`
+	ActiveVersion uint                          `mapstructure:"version"`
+	Versions      []*Version                    `mapstructure:"versions"`
+}
+
+// PlatformPrivateKeysAttributes .
+type PlatformPrivateKeysAttributes struct {
+	KeyLength     string     `mapstructure:"key_length"`
+	KeyType       string     `mapstructure:"key_type"`
+	Name          string     `mapstructure:"name"`
+	CreatedAt     *time.Time `mapstructure:"created_at"`
+	Replace       bool       `mapstructure:"replace"`
+	PublicKeySHA1 string     `mapstructure:"public_key_sha1"`
+}
+
+// GetPlatformPrivateKeys returns stats data based on GetPlatformPrivateKeysInput
+func (c *Client) GetPlatformPrivateKeys() (*PlatformPrivateKeysResponse, error) {
+
+	p := "/tls/private_keys"
+
+	r, err := c.Get(p, &RequestOptions{
+		Params: map[string]string{},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var ppkr *PlatformPrivateKeysResponse
+	if err := decodeJSON(&ppkr, r.Body); err != nil {
+		return nil, err
+	}
+
+	return ppkr, nil
+}
+
+// GetPlatformPrivateKey returns a specific private key
+func (c *Client) GetPlatformPrivateKey(i *GetPlatformPrivateKeyInput) (*PlatformPrivateKeyResponse, error) {
+
+	p := "/tls/private_keys"
+
+	if i.ID != "" {
+		p = fmt.Sprintf("%s/%s", p, i.ID)
+	}
+
+	r, err := c.Get(p, &RequestOptions{
+		Params: map[string]string{},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var ppkr *PlatformPrivateKeyResponse
+	if err := decodeJSON(&ppkr, r.Body); err != nil {
+		return nil, err
+	}
+
+	return ppkr, nil
+}
+
+// CreatePlatformPrivateKeyInput is an input to the CreatePlatformPrivateKey function.
+// Allowed values for the fields are described at https://docs.fastly.com/api/platform-tls
+type CreatePlatformPrivateKeyInput struct {
+	Data CreatePlatformPrivateKeyData `json:"data"`
+}
+
+// CreatePlatformPrivateKeyData .
+type CreatePlatformPrivateKeyData struct {
+	Type       string                             `json:"type"`
+	Attributes CreatePlatformPrivateKeyAttributes `json:"attributes"`
+}
+
+// CreatePlatformPrivateKeyAttributes .
+type CreatePlatformPrivateKeyAttributes struct {
+	Key  string `json:"key,omitempty"`
+	Name string `json:"name,omitempty"`
+}
```

Comment:
As above, let's just model the attributes and delete the JSONAPI structs. So `CreatePlatformPrivateKeyAttributes` should become `CreatePlatformPrivateKeyInput `